spec/plotting/vector_spec.rb

```diff
@@ -63,3 +63,30 @@
     end
   end
 end
+
+describe Daru::Vector, 'plotting category' do
+  let(:plot) { instance_double('Nyaplot::Plot') }
+  let(:diagram) { instance_double('Nyaplot::Diagram') }  
+  before do
+    allow(Nyaplot::Plot).to receive(:new).and_return(plot)
+  end
+  context 'bar' do
+    let(:dv) do
+      Daru::Vector.new ['III']*10 + ['II']*5 + ['I']*5,
+        type: :category,
+        categories: ['I', 'II', 'III']
+    end
+    subject do 
+      dv.plot(type: :bar) do |p, d|
+        p.x_label 'Categories'
+        p.y_label 'Frequency'
+      end
+    end
+      
+    it do
+      expect(plot).to receive(:add).with(:bar, ['I', 'II', 'III'], [5, 5, 10])
+      expect(plot).to receive(:show)
+      dv.plot(type: :bar)
+    end
```

Comment:
Not much. You can use [spies](https://relishapp.com/rspec/rspec-mocks/docs/basics/spies) and maybe convert it to something like:
```ruby
subject { spy(plot) }
before { dv.plot(type: :bar) }
it { is_expected.to have_received(:add).with(:bar, ['I', 'II', 'III'], [5, 5, 10]) }
```
...but I'm not sure this approach is flexible enough for all tests.